modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/SourceCallbackContextArgumentResolver.java

```diff
@@ -27,6 +28,7 @@
 
   @Override
   public SourceCallbackContext resolve(ExecutionContext executionContext) {
-    return ((ExecutionContextAdapter<ComponentModel>) executionContext).getVariable(LEGACY_SOURCE_CALLBACK_CONTEXT_PARAM);
+    return new LegacySourceCallbackContextAdapterAdapter(((ExecutionContextAdapter<ComponentModel>) executionContext)
```

Comment:
This is assuming that a legacy callback will always be obtained and should always be adapted. Leave this read to function when the new API so that we don't have to touch it again when the time comes